src/ProjectModel.php

```diff
@@ -169,26 +169,43 @@ public function update($pid, $model, $dryRun = false)
             if (count($update['lessDestructiveMaql'])) {
                 foreach ($update['lessDestructiveMaql'] as $i => $m) {
                     try {
+                        $m = $this->updateMaqlWithMultivalueAttributes($m, $multivalueAttributes);
                         $this->client->getDatasets()->executeMaql($pid, $m);
                     } catch (Exception $e) {
                         if (!empty($update['moreDestructiveMaql'][$i])) {
-                            $m = $update['moreDestructiveMaql'][$i];
+                            $m = $this->updateMaqlWithMultivalueAttributes($update['moreDestructiveMaql'][$i], $multivalueAttributes);
                             $this->client->getDatasets()->executeMaql($pid, $m);
                         } else {
                             throw $e;
                         }
                     }
-
                     return [
                         'description' => $update['description'],
-                        'maql' => $m
+                        'maql' => $m,
                     ];
                 }
             }
             return false;
         }
     }
 
+    private function updateMaqlWithMultivalueAttributes(string $maql, array $multivalueAttributes): string
+    {
+        $lines = explode("\n", $maql);
+
+        foreach ($multivalueAttributes as $attribute) {
+            foreach ($lines as &$line) {
+                if (strpos($line, sprintf('ALTER ATTRIBUTE {%s} ADD KEYS', $attribute)) === 0) {
+                    $line = str_replace(';', ' MULTIVALUE;', $line);
```

Comment:
nemůže být někde uprostřed středník?

Reply:
@ondrajodas Neměl by být nikde a v tom ALTER ATTRIBUTE nebude určitě, ty klíče si GD pojmenovává safe.